modules/core/src/encoder/field/PurposeRestrictionVectorEncoder.ts

```diff
@@ -60,10 +69,28 @@ export class PurposeRestrictionVectorEncoder {
 
           }
 
+          let isRangeEncodeRequired = i === len - 1;
+
+          if (!isRangeEncodeRequired) {
+
+            const {nextVendorId, index} = nextGvlVendor(vendorId, vendors[len - 1]);
+
+            if (vendors[i + 1] > nextVendorId) {
```

Comment:
vendors[i+1] is always defined ?

Reply:
yes, in fact if it is not defined the relevant block of code will not be executed. See the condition at line 74 and the condition at line 61